### ISSUE TITLE
bug: xlsx drops the first row of this file 

### ISSUE THREAD

**Issue by stla** [qrilka/xlsx#103]
Hello, 

 The first row of [this file](https://www.dropbox.com/s/3717azf93o4ayg3/bug.xlsx?dl=0) is not returned by `xlsx`:

```
Prelude X L CL> bs <- L.readFile "bug.xlsx"
Prelude X L CL> toXlsx bs ^? ixSheet "Sheet1" . ixCell (1,1) . cellValue
Just Nothing
Prelude X L CL> toXlsx bs ^? ixSheet "Sheet1" . ixCell (2,1) . cellValue
Just (Just (CellDouble 1887.0))
Prelude X L CL> toXlsx bs ^? ixSheet "Sheet1" . ixCell (1,2) . cellValue
Just Nothing
```

![capture](https://user-images.githubusercontent.com/4466543/31049420-d1daf56e-a632-11e7-8ba2-c5f775f1b364.PNG)

But if I open and save the file with Excel, then `xlsx` correctly read the new file. Looks like Excel corrects something in this file.

**Comment by qrilka** [comment]
Hi @stla what's the source of this file? It's the first time I see `inlineStr` cell value type being used :)
I've seen it listed in the spec but thought that it wasn't really used (like `d` for date type)
It shouldn't be very hard to add but I'm not sure how fast I could do that. Is it very critical for your code?

**Comment by stla** [comment]
Hi @qrilka ,
This file was created by the Javascript library `datatables`. You can try it here (the "Excel" button): https://datatables.net/extensions/buttons/examples/initialisation/export
It's a bit embarassing that `xlsx` cannot read such a file. But that's nice if you can fix that in the future, there's nothing urgent.

**Comment by stla** [comment]
This works if I replace 
`<c t="inlineStr" r="B2" s="2"><is><t>Position</t></is></c>`
with
`<c t="str" r="B2" s="2"><v>Position</v></c>`

So it should be easy to get this output by hacking the JS library.

**Comment by qrilka** [comment]
ok, thanks @stla

**Comment by qrilka** [comment]
BTW @stla that use of `str` type isn't quite correct - it's purpose is for formula results which give string result (and contain precomputed value)

**Comment by qrilka** [comment]
Resolved in 7e8d24c8c718ed8381cd4e53f4dada37ffb2d259
